src/test/java/com/easypost/AddressTest.java

```diff
@@ -0,0 +1,152 @@
+package com.easypost;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.List;
+
+import com.easypost.exception.EasyPostException;
+import com.easypost.model.Address;
+import com.easypost.model.AddressCollection;
+import com.easypost.model.Batch;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.BeforeAll;
+import static org.junit.jupiter.api.Assertions.*;
+
+
+public class AddressTest {
+    private static Address address;
+
+    /**
+     * Setup the testing environment for this file.
+     *
+     * @throws EasyPostException when the request fails.
+     */
+    @BeforeAll
+    public static void setup() throws EasyPostException{
+        EasyPost.apiKey = System.getenv("EASYPOST_TEST_API_KEY");
+
+        address = Address.create(Fixture.basicAddress());
+    }
+
+    /**
+     * Test creating an address.
+     *
+     * @throws EasyPostException when the request fails.
+     */
+    @Test
+    public void testCreate() throws EasyPostException {
+        Address address = Address.create(Fixture.basicAddress());
+
+        assertTrue(address instanceof Address);
+        assertEquals("388 Townsend St", address.getStreet1());
+        assertTrue(address.getId().startsWith("adr_"));
+    }
+
+    /**
+     * Test creating an address with verify_strict param.
+     *
+     * @throws EasyPostException when the request fails.
+     */
+    @Test
+    public void testCreateVerifyStrict() throws EasyPostException {
+        Map<String, Object> addressData = Fixture.basicAddress();
+        List<String> verificationList = new ArrayList<>();
+        
+        verificationList.add("true");
+        addressData.put("verify_strict", verificationList);
+
+        Address address = Address.create(addressData);
+
+        assertTrue(address instanceof Address);
+        assertTrue(address.getId().startsWith("adr_"));
+        assertEquals("388 TOWNSEND ST APT 20", address.getStreet1());
+    }
+
+    /**
+     * Test retrieving an address.
+     *
+     * @throws EasyPostException when the request fails.
+     */
+    @Test
+    public void testRetrieve() throws EasyPostException{
+        Address retrievedAddress = Address.retrieve(address.getId());
+
+        assertTrue(retrievedAddress instanceof Address);
+        assertTrue(retrievedAddress.getId().startsWith("adr_"));
+        assertEquals(address.getId(), retrievedAddress.getId());
```

Comment:
We should be asserting the entire retrieved object matches the entire created object instead of just the IDs. There's only a few objects where asserting the ID only makes sense (batches for instance because the object changes between requests as it moves through the queue and processes the batch).

Please go through the entire test suite and have the assertions match what the PHP library is as close as possible. If there is a reason we cannot assert against the entire object, call it out here so I'm made aware of it.